responsibleai/responsibleai/rai_insights/rai_insights.py

```diff
@@ -396,12 +446,31 @@ def _validate_rai_insights_input_parameters(
                         'The train labels and distinct values in '
                         'target (test data) do not match')
 
+            if feature_metadata is not None:
+                if not isinstance(feature_metadata, FeatureMetadata):
+                    raise UserConfigValidationException(
+                        "Expecting type FeatureMetadata but got {0}".format(
+                            type(feature_metadata)))
+
+                feature_metadata.validate_feature_metadata_with_user_features(
+                    list(train.columns))
+
             if model is not None:
                 # Pick one row from train and test data
-                small_train_data = train.iloc[0:1].drop(
-                    columns=[target_column])
-                small_test_data = test.iloc[0:1].drop(
-                    columns=[target_column])
+                small_train_data = train[0:1]
+                small_test_data = test[0:1]
+                if feature_metadata is not None:
+                    if feature_metadata.dropped_features is not None and \
+                            len(feature_metadata.dropped_features) != 0:
+                        small_train_data = train[0:1].drop(
```

Comment:
nit: you can just re-use small_train_data and small_test_data instead of calling train[0:1] and test[0:1] again here